src/clients/contracts/getters.ts

```diff
@@ -10,79 +12,161 @@ import vaiUnitrollerAbi from 'constants/contracts/abis/vaiUnitroller.json';
 import vaiVaultAbi from 'constants/contracts/abis/vaiVault.json';
 import xvsVaultStoreAbi from 'constants/contracts/abis/xvsVaultStore.json';
 import xvsVaultAbi from 'constants/contracts/abis/xvsVault.json';
-import vBepAbi from 'constants/contracts/abis/vBep.json';
+import vBep20Abi from 'constants/contracts/abis/vBep20.json';
 import vBnbTokenAbi from 'constants/contracts/abis/vBnbToken.json';
 import xvsTokenAbi from 'constants/contracts/abis/xvsToken.json';
 import venusLensAbi from 'constants/contracts/abis/venusLens.json';
-import governorBravoAbi from 'constants/contracts/abis/governorBravoDelegate.json';
+import governorBravoDelegateAbi from 'constants/contracts/abis/governorBravoDelegate.json';
 import xvsVestingAbi from 'constants/contracts/abis/xvsVesting.json';
 import vrtConverterAbi from 'constants/contracts/abis/vrtConverter.json';
 import vrtTokenAbi from 'constants/contracts/abis/vrtToken.json';
 import vrtVaultAbi from 'constants/contracts/abis/vrtVault.json';
+import {
+  Bep20,
+  Comptroller,
+  InterestModel,
+  Oracle,
+  VaiVault,
+  VaiUnitroller,
+  XvsVault,
+  XvsVaultStore,
+  VenusLens,
+  GovernorBravoDelegate,
+  XvsVesting,
+  VrtVault,
+  VrtConverter,
+} from 'types/contracts';
 import { getContractAddress, getToken, getVBepToken } from 'utilities';
+import { TokenContract, VTokenContract } from './types';
 
-const getContract = (abi: $TSFixMe, address: $TSFixMe, web3Instance: Web3) => {
+const getContract = (abi: AbiItem | AbiItem[], address: string, web3Instance: Web3) => {
   const web3 = web3Instance ?? getWeb3NoAccount();
   return new web3.eth.Contract(abi, address);
 };
 
-export const getTokenContract = (web3: Web3, name: $TSFixMe) => {
-  let abi = bep20Abi as $TSFixMe;
+export const getTokenContract = <T extends TokenId>(tokenId: T, web3: Web3): TokenContract<T> => {
+  const tokenAddress = getToken(tokenId).address;
 
-  if (name === 'xvs') {
-    abi = xvsTokenAbi;
-  } else if (name === 'vai') {
-    abi = vaiTokenAbi;
-  } else if (name === 'vrt') {
-    abi = vrtTokenAbi;
+  if (tokenId === 'xvs') {
+    return getContract(xvsTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
   }
 
-  return getContract(abi, getToken(name).address, web3);
+  if (tokenId === 'vai') {
+    return getContract(vaiTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
+  }
+
+  if (tokenId === 'vrt') {
+    return getContract(vrtTokenAbi as AbiItem[], tokenAddress, web3) as unknown as TokenContract<T>;
```

Comment:
I think we could improve this by making `getContract` generic

Reply:
It does not have a generic type. This is what I was thinking 
```
const getContract = <T extends TokenId>(abi: AbiItem | AbiItem[], address: string, web3Instance: Web3) => {
  const web3 = web3Instance ?? getWeb3NoAccount();
  return new web3.eth.Contract(abi, address) as unknown as TokenContract<T>;
};

export const getTokenContract = <T extends TokenId>(tokenId: T, web3: Web3): TokenContract<T> => {
  const tokenAddress = getToken(tokenId).address;

  if (tokenId === 'xvs') {
    return getContract<T>(xvsTokenAbi as AbiItem[], tokenAddress, web3);
  }

  if (tokenId === 'vai') {
    return getContract<T>(vaiTokenAbi as AbiItem[], tokenAddress, web3);
  }

  if (tokenId === 'vrt') {
    return getContract<T>(vrtTokenAbi as AbiItem[], tokenAddress, web3);
  }
  ```

Reply:
Oh I see, I didn't understand that. Works for me!